img-config/update-ssh-keys.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Verify that ssh config is correct before making any changes
+sudo sshd -t
+if [[ $? -ne 0 ]]; then
+    echo "sshd configuration is found to be incorrect before run, no changes done."
+    exit 1
+fi
+
+NOW=`date '+%F_%H:%M:%S'`
+OLD_DIR="/etc/ssh/keys_$NOW.old"
+
+# Creating the directory to store old keys
+sudo mkdir -p $OLD_DIR
+# Backup old keys
+sudo mv /etc/ssh/ssh_host*_key $OLD_DIR
+sudo mv /etc/ssh/ssh_host*_key.pub $OLD_DIR
+
+# This will generate new keys and place them in the default location
+sudo ssh-keygen -A
+sync
+
+# Check if ssh config is correct before restarting the service
+sudo sshd -t
+if [[ $? -ne 0 ]]; then
+    echo "sshd configuration check failed after key regeneration, old keys are in $OLD_DIR ."
+    echo "Please check the sshd configuration and restore the old keys if necessary."
+    echo "To restore the old keys, run the following commands:"
+    echo "sudo mv $OLD_DIR/*key /etc/ssh/"
+    exit 1
```

Comment:
Just being nitpicky, but could this step restore the files from `$OLD_DIR/` rather than leaving the task to the user?

While I find `ssh-keygen -A` very unlikely to fail, in the case something goes wrong, it would be a better end-user experience for the script to rollback automatically.

Reply:
I've added the fallback. What do you think of the changed version? I haven't tested this one on printer, but it should work, the changes weren't big.

Primary case I'm concerned here is no free space at all while running the script. In this case, everything might behave weirdly. So both the `ssh-keygen` and `cp` will fail.